src/lib/src/component/navigation/navigation.component.spec.ts

```diff
@@ -163,16 +163,16 @@ describe('NavigationComponent', () => {
   it('updates the UI state when an "navigation.deleted" event is received', async(() => {
     // given
     fixture.whenStable().then(() => {
-    component.getWorkspace().setDirty(tclFile.path, true);
-    component.getWorkspace().setExpanded(tclFile.path, true);
-    component.getWorkspace().setSelected(tclFile.path);
+    component.getWorkspace().setDirty(subfolder.path, true);
```

Comment:
Is there a reason why you changed this from 'tclFile' to 'subfolder'?

Reply:
yes, the original test assumption kinda only worked "by accident", considering that the test checks whether the element of type file, not folder, is expanded.

Previously, user actions to expand a node went through a method in NavigationComponent:

```
private expandNode(element: WorkspaceElement): void {
  if (element !== null && element.type === ElementType.Folder) {
    this.workspace.setExpanded(element.path, true);
  }
}
```

It only allows folders to be expanded, which makes sense, of course. I removed this method in favor of calling the `setExpanded` method of class `Workspace`, directly. I also moved the check for element type there, so now there is no way for this test to set a file type element into an invalid expanded state.